src/internal/operations/restore.go

```diff
@@ -228,6 +229,25 @@ func (op *RestoreOperation) do(
 		"restore_protected_resource_id", restoreToProtectedResource.ID(),
 		"restore_protected_resource_name", clues.Hide(restoreToProtectedResource.Name()))
 
+	// Check if the resource has the service enabled to be able to restore.
+	runnable, err := op.rc.IsServiceEnabled(
+		ctx,
+		op.Selectors.PathService(),
+		restoreToProtectedResource.ID())
+	if err != nil {
+		logger.CtxErr(ctx, err).Error("verifying restore is runnable")
+		op.Errors.Fail(clues.Wrap(err, "verifying restore is runnable"))
+
+		return nil, clues.Stack(err).WithClues(ctx)
+	}
+
+	if !runnable {
+		logger.CtxErr(ctx, graph.ErrServiceNotEnabled).Error("checking if restore is enabled")
+		op.Errors.Fail(clues.Wrap(err, "checking if restore is enabled"))
+
+		return nil, clues.Stack(graph.ErrServiceNotEnabled).WithClues(ctx)
```

Comment:
same duplication with these: they can be dropped, since the calling method will handle it.  Then the returned error just needs to get augmented.
```suggestion
		return nil, clues.Wrap(graph.ErrServiceNotEnabled, "verifying service is enabled for restore").WithClues(ctx)
```